Update R installation docs, libgit2 required for devtools dependency

Following the R installation docs
```bash
$ R -e "install.packages('devtools', repos='http://cran.rstudio.com/')"
...
checking whether the libgit2 version will work in git2r... no
----- Results of the git2r package configure -----

  PKG_CFLAGS:
  PKG_LIBS: -lgit2

--------------------------------------------------
configure: error: in `/private/var/folders/m6/4yqn_4q129lbth_dq3qzj_8h0000gn/T/RtmpXH84bv/R.INSTALL946f755c6779/git2r':
configure: error:
  ---------------------------------------------
   The libgit2 library that is required to build git2r was not found.

   Please install:
     libgit2-dev   (package on e.g. Debian and Ubuntu)
     libgit2-devel (package on e.g. Fedora, CentOS and RHEL)
     libgit2       (Homebrew package on macOS)
   and try again.

   If the libgit2 library is installed on your system but the git2r configuration is
   unable to find it, you can specify the include and lib path to libgit2 with:
   R CMD INSTALL git2r --configure-vars='INCLUDE_DIR=path/to/include LIB_DIR=path/to/lib'
  ---------------------------------------------
See `config.log' for more details
ERROR: configuration failed for package ‘git2r’
```

On OSX using homebrew worked for me
```
$ brew install libgit2
Updating Homebrew...
==> Auto-updated Homebrew!
Updated 2 taps (homebrew/cask and homebrew/core).
==> Updated Formulae
mariadb@10.2

==> Installing dependencies for libgit2: libssh2
==> Installing libgit2 dependency: libssh2
==> Downloading https://homebrew.bintray.com/bottles/libssh2-1.8.0.sierra.bottle.1.tar.gz
######################################################################## 100.0%
==> Pouring libssh2-1.8.0.sierra.bottle.1.tar.gz
🍺  /usr/local/Cellar/libssh2/1.8.0: 182 files, 799KB
==> Installing libgit2
==> Downloading https://homebrew.bintray.com/bottles/libgit2-0.27.7.sierra.bottle.tar.gz
######################################################################## 100.0%
==> Pouring libgit2-0.27.7.sierra.bottle.tar.gz
🍺  /usr/local/Cellar/libgit2/0.27.7: 106 files, 3.2MB

$ R -e "install.packages('devtools', repos='http://cran.rstudio.com/')"
...
*** installing help indices
*** copying figures
** building package indices
** installing vignettes
** testing if installed package can be loaded
* DONE (devtools)
```

--- a/docs/installation/source.rst
+++ b/docs/installation/source.rst
@@ -124,7 +124,7 @@ ADAM supports SparkR, for Spark 2.1.0 and onwards. To build and test
 
 .. code:: bash
 
-    mvn -Pr package
+    mvn -P r package
 
 This will enable the ``adam-r`` module as part of the ADAM build. This
 module uses Maven to invoke the ``R`` executable to build the
@@ -137,9 +137,21 @@ module uses Maven to invoke the ``R`` executable to build the
     R -e "install.packages('roxygen2', repos='http://cran.rstudio.com/')"
     R -e "install.packages('devtools', repos='http://cran.rstudio.com/')"
 
+Installation of ``devtools`` may require ``libgit2`` as a dependency.
+
+.. code:: bash
+
+    apt-get install libgit2-dev
+
 The build also requires you to have the ``SparkR`` package installed,
-and the ADAM JARs must be built and provided to ``SparkR``. This can be
-done with the following bash commands:
+where ``v2.x.x`` should match your Spark version.
+
+.. code:: bash
+
+   R -e "devtools::install_github('apache/spark@v2.x.x', subdir='R/pkg')"
+
+The ADAM JARs can then be provided to ``SparkR`` with the following bash
+commands:
 
 .. code:: bash
 

